src/libraries/System.Private.CoreLib/src/System/Collections/ObjectModel/ReadOnlyCollection.cs

```diff
@@ -229,4 +230,34 @@ void IList.RemoveAt(int index)
             ThrowHelper.ThrowNotSupportedException(ExceptionResource.NotSupported_ReadOnlyCollection);
         }
     }
+
+    /// <summary>
+    /// Provides static methods to create read-only collections and sets.
+    /// </summary>
+    public static class ReadOnlyCollection
+    {
+        /// <summary>
+        /// Creates a new <see cref="ReadOnlyCollection{T}"/> from the specified span of values.
+        /// This method (simplifies collection initialization)[/dotnet/csharp/language-reference/operators/collection-expressions]
+        /// to create a new <see cref="ReadOnlyCollection{T}"/> with the specified values.
+        /// </summary>
+        /// <typeparam name="T">The type of elements in the collection.</typeparam>
+        /// <param name="values">The span of values to include in the collection.</param>
+        /// <returns>A new <see cref="ReadOnlyCollection{T}"/> containing the specified values.</returns>
+        [Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0301:Simplify collection initialization", Justification = "This method simplifies ReadOnlyCollection<T> initialization, so you cannot simplify collection initialization inside it.")]
```

Comment:
We typically suppress warnings using `#pragma` declarations. What necessitates putting a suppress attribute on the implementation?

Reply:
This doesn't need to be suppressed at all. It's not a warning, only a suggestion.

Reply:
> This doesn't need to be suppressed at all. It's not a warning, only a suggestion.

I suppressed this suggestion to prevent it from being accidentally applied to the code because it would cause recursion when it was compiled. I can remove the suppression.